narwhals/_exceptions.py

```diff
@@ -5,3 +5,14 @@ class ColumnNotFoundError(Exception): ...
 
 
 class InvalidOperationError(Exception): ...
+
+
+class InvalidIntoExprError(TypeError):
```

Comment:
subclassing TypeError in case some downstream library is already catching this. (Also because it is actually an error about types)